### PR TITLE
add read_event_from_json function to pyghee.lib

### PR DESCRIPTION
**Pull request by trz42**
Moved read_event_from_json from EESSI/eessi-bot-software-layer/eessi_bot_software_layer.py to PyGHee/pyghee/lib.py

### CODE REVIEW

**Comment by trz42** [comment]
There is a bit overlap with PR#2 (using X-Github-Delivery instead of X-Request-Id).

File: pyghee/lib.py [hunk 1]
```diff
@@ -47,6 +49,18 @@ def get_event_info(request):
...
 
 
+def read_event_from_json(jsonfile):
+    """
+    Read event data from a json file.
+    """
+    req = namedtuple('Request', ['headers', 'json'])
+    with open(jsonfile, 'r') as jf:
+        event_data = json.load(jf)
+        req.headers = CaseInsensitiveDict(event_data['headers'])
```
Comment (boegel) [inline pyghee/lib.py:59]: @trz42 Why use `CaseInsensitiveDict` here?

Comment (boegel) [reply pyghee/lib.py:59]: Ah, maybe this is a question for @bedroge or @sassy-crick

Comment (trz42) [reply pyghee/lib.py:59]: I just copied/moved it as is from EESSI/eessi-bot-software-layer/eessi_bot_software_layer.py

Comment (boegel) [reply pyghee/lib.py:59]: OK. Not sure I like the case-insensitive aspect of it though...

@bedroge @sass-crick What was the reasoning behind this move?
Avoiding `KeyError` due to incorrect case in a key value?

Comment (bedroge) [reply pyghee/lib.py:59]: Yes, I remember I ran into a weird issue related to the case of a key, I think it was something like `X-Github-Delivery` vs `X-GitHub-Delivery`. Don't know all the details anymore, but I think the original file was a dump of an event captured by Smee, while I ran into the error when actually testing with a real PR. I was quite surprised to see that, maybe GitHub changed something along the way (?).

**Pull request merged**